overlay/sbin/k3os-upgrade

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+set -x
+
+PROC=$(uname -m)
+
+# Take a parameter of the version number (i.e. v0.4.0) if it is given, otherwise use latest
+if [ -z $K3OS_VERSION ]
+then
+	K3OS_VERSION="latest"
+fi
+
+if [ $PROC == "x86_64" ]
+then
+	ARCH="amd64"
+elif [ $PROC == "aarch64" ]
+then
+	ARCH="arm64"
+elif [ $PROC == "armv7l" ]
+then
+	ARCH="arm"
+else
+	echo "Unsupported CPU architecture."
+	exit 0
+fi
+
+echo "Upgrading k3os to ${VERSION}"
+
+cd /k3os/system/
+mount -o remount rw .
+curl -LO https://github.com/rancher/k3os/releases/download/${K3OS_VERSION}/k3os-rootfs-${ARCH}.tar.gz
+tar xfz k3os-rootfs-${ARCH}.tar.gz; sudo cp -R ${K3OS_VERSION}/k3os/system/* .
+rm -rf k3os-rootfs-${ARCH}.tar.gz ${K3OS_VERSION}
```

Comment:
Maybe something like this is cleaner:
```
K3OS_ROOTFS_URL="https://github.com/rancher/k3os/releases/download/${K3OS_VERSION}/k3os-rootfs-${ARCH}.tar.gz"
curl -fsSL "${K3OS_ROOTFS_URL}" | tar xz --strip-components=3
```

Reply:
Would also prefer this be renamed to `k3os-upgrade-rootfs`

Reply:
The current code won't work actually as when you download from "latest" the extracted files will be under a specific version number folder and not a folder named "latest".

Reply:
Cool, adopted these changes.